packages/config/src/reader.ts

```diff
@@ -280,10 +280,26 @@ export class ConfigReader implements Config {
 
   /** {@inheritdoc Config.getOptionalBoolean} */
   getOptionalBoolean(key: string): boolean | undefined {
-    return this.readConfigValue(
+    const value = this.readConfigValue<string | boolean>(
       key,
-      value => typeof value === 'boolean' || { expected: 'boolean' },
+      val =>
+        typeof val === 'boolean' ||
+        typeof val === 'string' || { expected: 'boolean' },
     );
+    if (typeof value === 'boolean' || value === undefined) {
+      return value;
+    }
+    let boolean;
```

Comment:
Thinking that we should probably go a little bit further and take some inspiration from the `yn` package and use some regex's like these https://github.com/sindresorhus/yn/blob/main/index.js#L17-L23 to match a few other values and do the coercion to boolean here.

Reply:
Probably don't want to bring the whole package in, but thinking we can lift the regex's out of that package into here instead.

Reply:
I don't think that it would be a good idea. People might want to keep `yes/no` as it is. We also don't want to coerce `0` to false as integer likes are already coerced to numbers. The boolean true and false are the only weird edge cases where they are valid tokens in javascript but can't be represented as such in environment variables.

Reply:
`yes/no` would be kept as is if they're requested with `getString` or `getOptionalString`, but in this case, theres a requirement to coerce the value that is set in config into the type that you're requesting. I think it's pretty reasonable to assume also that if I have in config something like this:

```yaml
enabled: ${IS_ENABLED}
```

With `IS_ENABLED=1` as an Environment Variable, If I'm doing `getBoolean('enabled')` i'd expect it to return `true`, but `getNumber('enabled')` I'd expect 1. The coercion is only happening in the `getOptionalBoolean` part which I think is fine.

Reply:
Alright, I am pushing a commit for the same.